### PR TITLE
Adjust ILM policy for deprecation logs

### PR DESCRIPTION
**Pull request by jakelandis**
This commit introduces a delete phase for deprecation logs. 
This commit also adjusts the rollover criteria such that there
will always be 30-60 days worth of deprecation logs.

### CODE REVIEW

**Comment by cjcenizal** [comment]
Seems like a great idea! What will the name of this policy be, once it's created?

File: x-pack/plugin/core/src/main/resources/org/elasticsearch/xpack/deprecation/deprecation-indexing-ilm-policy.json [hunk 1]
```diff
@@ -4,9 +4,18 @@
...
     }
+  },
+  "delete": {
+    "min_age": "30d",
+    "actions":{
+      "delete": {}
+    }
+  },
+  "_meta": {
+    "description": "ILM policy used for deprecation logs"
```
Comment (dakrone) [inline x-pack/plugin/core/src/main/resources/org/elasticsearch/xpack/deprecation/deprecation-indexing-ilm-policy.json:19]: Do we also want "managed": true for this policy?

**Review by dakrone — APPROVED** [review]
LGTM, I left one question, no need for another review regardless of the answer

File: x-pack/plugin/core/src/main/resources/org/elasticsearch/xpack/deprecation/deprecation-indexing-ilm-policy.json (hunk 1 above)
Comment (jakelandis) [reply x-pack/plugin/core/src/main/resources/org/elasticsearch/xpack/deprecation/deprecation-indexing-ilm-policy.json:19]: I think we want users to be able to easily modify this one, but open to counter opinion.

**Comment by jakelandis** [comment]
> Seems like a great idea! What will the name of this policy be, once it's created?  

The policy already exists as '.deprecation-indexing-ilm-policy' , this PR just adds the delete and adjusts the rollover.

File: x-pack/plugin/core/src/main/resources/org/elasticsearch/xpack/deprecation/deprecation-indexing-ilm-policy.json (hunk 1 above)
Comment (jakelandis) [reply x-pack/plugin/core/src/main/resources/org/elasticsearch/xpack/deprecation/deprecation-indexing-ilm-policy.json:19]: on second thought ... maybe it should be managed we don't want to encourage deletion and is associate with a hidden data stream. (and the name starts with a `.`) 
thoughts ?

Comment (dakrone) [reply x-pack/plugin/core/src/main/resources/org/elasticsearch/xpack/deprecation/deprecation-indexing-ilm-policy.json:19]: +1 to have as managed, it's more of a suggestion anyway :)

**Comment by jakelandis** [comment]
@elasticmachine update branch

**Pull request merged**